ontolearn/concept_learner.py

```diff
@@ -1623,20 +1668,39 @@ def fit_from_iterable(self, dataset: Union[List[Tuple[str, Set[OWLNamedIndividua
             print("Predictions: ", predictions_str)
         return predictions_as_owl_class_expressions
 
-    def train(self, data: Iterable[List[Tuple]], epochs=300, batch_size=None, learning_rate=1e-4, decay_rate=0.0,
+    @staticmethod
+    def generate_training_data(kb_path, num_lps=1000, storage_dir="./NCES_Training_Data"):
+        lp_gen = LPGen(kb_path=kb_path, max_num_lps=num_lps, storage_dir=storage_dir)
+        lp_gen.generate()
+        print("Loading generated data...")
+        with open(f"{storage_dir}/LPs.json") as file:
+            lps = list(json.load(file).items())
+            print("Number of learning problems:", len(lps))
+        return lps
+
+
+
+    def train(self, data: Iterable[List[Tuple]]=None, epochs=50, batch_size=64, num_lps=1000, learning_rate=1e-4, decay_rate=0.0,
               clip_value=5.0, num_workers=8, save_model=True, storage_path=None, optimizer='Adam', record_runtime=True,
               example_sizes=None, shuffle_examples=False):
+        if os.cpu_count() <= num_workers:
+            num_workers = max(0,os.cpu_count()-1)
+        if storage_path is None:
+            currentDateAndTime = datetime.now()
+            storage_path = f'NCES-Experiment-{currentDateAndTime.strftime("%H:%M:%S")}'
```

Comment:
The colon character `:` is not allowed in Windows filenames and therefore an error will occur in this systems.
I'll change this to the following:
```suggestion
            storage_path = f'NCES-Experiment-{currentDateAndTime.strftime("%H-%M-%S")}'
```